tools/accuracy_checker/accuracy_checker/evaluators/quantization_model_evaluator.py

```diff
@@ -333,18 +333,22 @@ def create_dataset_attributes(config, tag):
     data_reader_config = dataset_config.get('reader', 'opencv_imread')
     data_source = dataset_config.get('data_source')
 
-    if isinstance(data_reader_config, str):
-        data_reader = BaseReader.provide(data_reader_config, data_source)
-    elif isinstance(data_reader_config, dict):
-        data_reader = BaseReader.provide(data_reader_config['type'], data_source, data_reader_config)
-    else:
-        raise ConfigError('reader should be dict or string')
     annotation_reader = None
     dataset_meta = {}
     metric_dispatcher = None
+
     if contains_any(dataset_config, ['annotation', 'annotation_conversion']):
         annotation_reader = Dataset(dataset_config)
         dataset_meta = annotation_reader.metadata
+
+    if isinstance(data_reader_config, str):
+        data_reader = BaseReader.provide(data_reader_config, data_source)
+    elif isinstance(data_reader_config, dict) and annotation_reader:
```

Comment:
hmm.. dictionary used not only for bert on our side.
I think you need to check reader type in your case and that annotation reader provided inside `elif` statement and throw appropriate error if for this reader type annotation_reader is None

Reply:
I mean, something like that:
```python
    elif isinstance(data_reader_config, dict):
       data_reader_type = data_reader_config['type']
       if data_reader_type == `annotation_features_extractor`:
           if not annotation_reader:
               raise ConfigError('annotation required')
           data_reader = BaseReader.provide(data_reader_config['type'], data_source, data_reader_config, annotation_reader.annotation)
       else:
           data_reader = BaseReader.provide(data_reader_config['type'], data_source, data_reader_config)
             
```

Reply:
But better define list of constants in data_reader.py , it will include reader providers which require annotation and check that reader not in this list instead direct string comparison

Reply:
@dliang0406, can I suggest my version of bugfix instead of your in separate PR?

Reply:
Hi @eaidova I think your bugfix is better since it treats "annotation_features_extractor" specifically and leave other readers untouched. But I don't get your point of  separated PR...
If I submit a new patch in this PR, you can squash the three commits together at your side, right?

Reply:
I mean that, this my patch is pretty simple for avoiding extra changes in your fix, but I want made more common approach for such situation which also requires changes in other files: `data_reader.py`, `model_evaluator.py`. It is the reason why I want submit another patch which with more general solution.

Reply:
Hi @eaidova , as I 'm not so familiar with data_reader.py, I think I can wait for your real fix and use my patch as a WA locally.